Bindings/Java/OpenSimJNI/Test/testContext.cpp

```diff
@@ -223,21 +223,29 @@ int main()
     std::cout << pathAfterTypeChangeToViaInXML << endl;
  
     // Make a change to a socket that is invalid and verify that we can recover
-    // from that invalid change by restoring from a cached model and state.
+    // from that invalid change by not making it on model directly
     context->cacheModelAndState();
     Joint& shoulder = model->updComponent<Joint>("r_shoulder");
     AbstractSocket& socket = shoulder.updSocket("child_frame");
     const std::string originalConnecteeName = socket.getConnecteeName();
     try {
         // create an invalid model where joint connects two frames on ground
-        socket.setConnecteeName("ground"); 
-        context->restoreStateFromCachedModel();
+        context->setConnecteeName(shoulder, socket, "ground");
     }
-    catch (...) {
-        // undo the change
-        socket.setConnecteeName(originalConnecteeName);
-        context->restoreStateFromCachedModel();
+    catch (const std::exception& e) {
+        // Expect meaningful error message rather than a crash
+        cout << "Exception: " << e.what() << endl;
```

Comment:
Agreed.

Reply:
@aseth1 we don't manufacture new custom messages, it's whatever the opensim-core throws. This message propagates to the GUI as an error dialog. The statement sets the expectations rather than make up a new message! If this can be clarified more let me know.

Reply:
Then I am not sure what you agreed with above.  *opensim-core* cannot does not know that the model in the GUI remained unchanged. In the GUI are you just going to pop-up a message that the `MultibodyGraphMaker` failed to build a valid tree (if that is the low level message)? It seems at the very least you could have `OpenSimContext` `try-catch` the `initSystem()` failure to provide the context that the user's attempted edit failed due to provided details (as describe below) and that model remained unchanged.